tools/flaky-test-reporter/github_issue.go

```diff
@@ -0,0 +1,460 @@
+			break
+		}
+	}
+	if nil == targetComment {
+		return nil, fmt.Errorf("no comment match")
+	}
+	return targetComment, nil
+}
+
+// getFlakyIssues filters all issues by flakyLabel, and return map {testName: slice of issues}
+// Fail if find any issue with no discoverable identifier(testIdentifierPattern missing),
+// also fail if auto comment not found.
+// In most cases there is only 1 issue for each testName, if multiple issues found open for same test,
+// most likely it's caused by old issues being reopened manually, in this case update both issues.
+func getFlakyIssues() (map[string][]*flakyIssue, error) {
+	issuesMap := make(map[string][]*flakyIssue)
+	reposForIssue, err := ghutil.ListRepos(org)
+	if nil != err {
+		return nil, err
+	}
+	for _, repoForIssue := range reposForIssue {
+		issues, err := ghutil.ListIssuesByRepo(org, repoForIssue, []string{flakyLabel})
+		if nil != err {
+			return nil, err
+		}
+		for _, issue := range issues {
+			if nil != issue.ClosedAt && issue.ClosedAt.Before(timeConsiderOld) {
+				continue // Issue closed long time ago, it might fail with a different reason now.
+			}
+			issueIdentity := regexp.MustCompile(reTestIdentifier).FindStringSubmatch(issue.GetBody())
+			if len(issueIdentity) < 2 { // Malformed issue, all auto flaky issues need to be identifiable.
+				return nil, fmt.Errorf("cannot get test identifier from auto flaky issue '%v'", err)
+			}
+			autoComment, err := findExistingComment(issue, issueIdentity[1])
+			if nil != err {
+				return nil, fmt.Errorf("cannot find auto comment for issue '%s': '%v'", *issue.URL, err)
+			}
+			issuesMap[issueIdentity[1]] = append(issuesMap[issueIdentity[1]], &flakyIssue{
+				issue: issue,
+				identity: &issueIdentity[1],
+				comment: autoComment,
+			})
+		}
+	}
+	// Handle test with multiple issues associated
+	// if all open: update all of them
+	// if all closed: only keep latest one
+	// otherwise(these may have been closed manually): remove closed ones from map
+	for k, v := range issuesMap {
+		hasOpen := false
+		hasClosed := false
+		for _, fi := range v {
+			switch fi.issue.GetState() {
+				case string(ghutil.IssueOpenState):
+					hasOpen = true
+				case string(ghutil.IssueCloseState):
+					hasClosed = true
+			}
+		}
+		fmt.Sprintln(k, hasOpen, hasClosed)
+		if hasOpen && hasClosed {
+			for i, fi := range v {
+				if string(ghutil.IssueCloseState) == fi.issue.GetState() {
+					issuesMap[k] = append(issuesMap[k][:i], issuesMap[k][i+1:]...)
+				}
+			}
+		} else if ! hasOpen {
+			sort.Slice(issuesMap[k], func(i, j int) bool {
+				return nil != issuesMap[k][i].issue.CreatedAt && 
+				( nil == issuesMap[k][j].issue.CreatedAt || issuesMap[k][i].issue.CreatedAt.Before(*issuesMap[k][j].issue.CreatedAt))
+			})
+			issuesMap[k] = []*flakyIssue{issuesMap[k][0]}
+		}
+	}
+	return issuesMap, err
+}
+
+func processGithubIssueForRepo(rd *RepoData, flakyIssuesMap map[string][]*flakyIssue, dryrun *bool) ([]string, error) {
```

Comment:
Would be nice to document this function, specially the return values.

Reply:
done